contrib: Autodetect GITHUB_TOKEN during release

joestringer opened the pull request:
If the GITHUB_TOKEN variable is not set, try to create it via local CLI.

joestringer committed:
contrib: Autodetect GITHUB_TOKEN during release

If the GITHUB_TOKEN variable is not set, try to create it via local CLI.

Signed-off-by: Joe Stringer <joe@cilium.io>

bimmlerd commented on contrib/release/prep-changelog.sh:
```diff
@@ -54,6 +54,7 @@ main() {
     local ersion="$(echo $2 | sed 's/^v//')"
     local version="v$ersion"
     local old_branch="$(echo $3 | sed 's/^v//')"
+    local GITHUB_TOKEN=${GITHUB_TOKEN:-"$(gh auth token)"}
```
What happens if `gh auth login` hasn't been executed prior?

joestringer replied:
In that case, `gh auth login` will print `no oauth token` and exit nonzero. Since this is executed within a subshell, the exit code does not result in the outer script failing. It'll continue, just like it'll do today with no github token defined, then fail some time later in the script with an authentication error.

Perhaps these scripts could all do with a `if ! gh auth status >/dev/null; then common:exit ... ; fi` snippet to ensure this will pass..?

joestringer replied:
^ Implemented the above, let me know what you think.

bimmlerd replied:
makes sense to me :+1:

joestringer committed:
contrib: Require github authentication for release

Adding this check at the start of the release process, and in the
prep-changelog script which will later execute $(gh auth token) if a
token is not available.

Signed-off-by: Joe Stringer <joe@cilium.io>
